builtin/credential/aws/pkcs7/pkcs7_test.go

```diff
@@ -217,26 +194,6 @@ func createTestCertificateByIssuer(name string, issuer *certKeyPair, sigAlg x509
 		case *dsa.PrivateKey:
 			template.SignatureAlgorithm = x509.DSAWithSHA256
 		}
-	case x509.DSAWithSHA1:
-		var dsaPriv dsa.PrivateKey
-		params := &dsaPriv.Parameters
-		err = dsa.GenerateParameters(params, rand.Reader, dsa.L1024N160)
-		if err != nil {
-			return nil, err
-		}
-		err = dsa.GenerateKey(&dsaPriv, rand.Reader)
-		if err != nil {
-			return nil, err
-		}
-		switch issuerKey.(type) {
-		case *rsa.PrivateKey:
-			template.SignatureAlgorithm = x509.SHA1WithRSA
-		case *ecdsa.PrivateKey:
-			template.SignatureAlgorithm = x509.ECDSAWithSHA1
-		case *dsa.PrivateKey:
-			template.SignatureAlgorithm = x509.DSAWithSHA1
-		}
-		priv = &dsaPriv
 	}
```

Comment:
I wonder if there should be a default case here that throws an error if `sigAlg` isn't one of the known cases?

Reply:
Possibly. But not critical I think since this is only in a `_test.go` file.